packages/devtools_app/lib/src/primitives/utils.dart

```diff
@@ -778,6 +778,19 @@ enum SortDirection {
   descending,
 }
 
+class LineRange extends Range {
+  const LineRange(super.begin, super.end);
+
+  @override
+  int get begin => super.begin.toInt();
+
+  @override
+  int get end => super.end.toInt();
+
+  @override
+  int get size => end - begin + 1;
```

Comment:
why is this Range now inclusive instead of being exclusive like I assume the normal Range class is?

Reply:
Basically for the `lineCount` argument to `Gutter`:

```dart
lineCount: widget.lineRange?.size ?? lines.length,
```

Maybe we should just make `LineRange` completely disjoint from `Range`?

Reply:
Making it disjoint sounds fine to me. I would also consider making the end value just be exclusive rather than inclusive as that is the default expectation for ranges.